src/mpl/src/timer/mpl_timer_clock_gettime.c

```diff
@@ -34,6 +34,14 @@ int MPL_wtime_touint(MPL_time_t * t, unsigned int *val)
     return MPL_SUCCESS;
 }
 
+int MPL_wtime_toll(MPL_time_t * t, long long int *val)
+{
+    *val = t->tv_sec * 1000000000;
+    *val += t->tv_nsec;
```

Comment:
So the integer timestamp unit is in `ns`? This need be documented wherever we introduce integer timestamp and maybe reference the Tool specification as well.

Reply:
There is no specified unit of time. According to the standard, section 15.3.8:
```
A timestamp in this context is a count of clock
ticks elapsed since some time in the past and represented as a variable of
type \const{MPI\_Count}.
```

Reply:
Is there a standardized interpreation of what is a `clock tick`?

Reply:
If the standard really means clock cycles, then some of the timer implementation where we use `ns` or `us` is questionable.

Reply:
Can we *only* enable the timestamp when cpu cycle timer is available?

Reply:
The MPI standard does not specify what constitutes a clock tick. It is not required to be clock cycles. Users can query for `ticks_per_second` of a given source to get an understanding of the resolution.

Reply:
> Users can query for `ticks_per_second` of a given source to get an understanding of the resolution.

Do we need a MPL function for this? Note that `MPL_wtick` does not return information relevant to `ticks_per_second` since what `MPL_wtick` returns is a resolution, but not the unit.

Reply:
Yes, a function is probably a good idea. We could just do `1.0 / tickval`, but that's maybe not the most accurate way to get the value.

Reply:
In the case of `clock_gettime`, since your unit is 1 nsec, your `ticks_per_second` is 10^9, not `1.0/MPL_wtick()`. Unit and resolution are not the same and can be completely unrelated.

Reply:
Hm, yeah. Now I'm seeing more problems with the variable naming of the existing functions.

It's easy enough to report ticks_per_second for the system timers that have known units. For cycle timers, we may just return `MPI_T_ERR_NOT_SUPPORTED`.